app/controllers/users_controller.rb

```diff
@@ -1,10 +1,14 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :admin_user, only: :destroy
+  before_action :load_user, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @users = User.page(params[:page]).per Settings.show_limit.show_10
+  end
 
   def show
-    @user = User.find_by id: params[:id]
-    return if @user
-    flash[:danger] = t "static_pages.home.error_messages"
-    redirect_to root_path
   end
 
   def new
```

Comment:
For 7 public methods in controller, please rearrange them follow order of object's life cycle: `index, new, create, show, edit, update, destroy`